Popover improvements #637, #656

ChronosSF opened the pull request:
Closes #637 , #656  .  

Exposing direction priority as an option. Enforcing direction when user-specified.

kdinev commented on src/js/modules/infragistics.ui.popover.js:
```diff
@@ -162,38 +166,38 @@
 			$( window ).on( "resize.popover", $.proxy( this._resizeHandler, this ) );
 		},
 		_createWidget: function (options, element) {
+			// initialization performance will be better if we don't try to normalize the defaults
```
Options should not be cached in an internal property. If you're changing the value of the option, then assign it back to the option. This causes a lot of issues long term.

kdinev commented on tests/unit/popover/common/tests.html:
```diff
@@ -473,7 +473,8 @@
                 ok( $( '#notifier_popover_arrow' ).css( "display" ) !== "none", 'Popover arrow not shown!' );
                 ok( $( '#notifier_popover_arrow' ).hasClass( "ui-igpopover-arrow-bottom" ), 'Popover arrow is not pointing to bottom!' );
-                ok( $( '#notifier_popover' ).offset().top + $( '#notifier_popover' ).height() <= $( '#notifier' ).offset().top,
+                ok( $( '#notifier_popover' ).offset().top + $( '#notifier_popover' ).height() <= $( '#notifier' ).offset().top || 
```
There is quite a few changes in the logic, and only one modification in the unit tests. Please write exhaustive unit tests for the new changes.

ChronosSF commented:
@kdinev , I removed the priority cache and instead I am assigning back the "fixed" prio array to the option. Also added tests unique to the new option. The other logic changes are pretty well covered by the remaining tests (had to change a few due to the new overlap enforcement in some cases).
